### ISSUE TITLE
Incorrect behavior after home/end key trigger

### ISSUE THREAD

**Issue by unbrokenclayman** [lukehaas/Scrollify#368]
Let's say we have 10 slides and scroll to the last, then press Home, getting to the first one, but now if we do scrolling upside, we will get to the 9th slide.
Same in reverse direction.

**Comment by prototype23** [comment]
@unbrokenclayman you can use this code to fix that problem:
`  $(window)
    .off('keyup.myNamespace')
    .on('keyup..myNamespace', function (e) {
      if (e && e.which) {
        switch (e.which) {
        case 35:
        case 36:
          setTimeout(function () {
            scrollify.update();
          }, 10);
          break;
        default:
        }
      }
    });`

**Comment by unbrokenclayman** [comment]
> @unbrokenclayman you can use this code to fix that problem:
> ` $(window) .off('keyup.myNamespace') .on('keyup.myNamespace', function (e) { if (e && e.which) { switch (e.which) { case 35: case 36: setTimeout(function () { scrollify.update(); }, 10); break; default: } } });`

That way doesn't working for me and causes even more problems: after I press home/end keys several times it may stops somwhere at the middle of page.

**Comment by prototype23** [comment]
Do you have an animation durion the scroll? The setTimeout function on my example must have longer duration than the scroll duration.

Then you could try to prevent the default action of home/end keys.

**Comment by unbrokenclayman** [comment]
> Do you have an animation on the scroll? The setTimeout function on my example must have longer duration than the scroll duration.

Nope, just native chrome scrolling. I trying to set time around 1000ms, which should be more than enough, but without success.

> If still you have issues, you could try to prevent the default action of home/end keys.

Why should'nt plugin set active slide properly itself after scrolling to top or bottom?

**Comment by prototype23** [comment]
I'm not the creator of this plugin :D

**Comment by unbrokenclayman** [comment]
Adding that code at the end of plugin keyHandler function helps:

```
} else if(e.keyCode==36) {
  e.preventDefault();
  index = 0;
  move(index, false);
} else if(e.keyCode==35) {
  e.preventDefault();
  index = heights.length-1;
  move(index, false);
}
```

**Comment by unbrokenclayman** [comment]
Ah, there already was a pull request #203
